feature-libs/my-account/organization/components/unit/services/unit-list.service.ts

```diff
@@ -16,61 +118,88 @@ import { OrganizationTableType } from '../../shared/organization.model';
 })
 export class UnitListService extends OrganizationListService<B2BUnit> {
   protected tableType = OrganizationTableType.UNIT;
-
+  protected treeAction$ = new BehaviorSubject<TreeAction>(
+    new CollapseFromLevelAction(1, this)
+  );
   constructor(
     protected tableService: TableService,
-    protected unitService: OrgUnitService
+    protected unitService: OrgUnitService,
+    protected unitItemService: UnitItemService
   ) {
     super(tableService);
+    this.unitItemService.key$
+      .pipe(
+        take(1),
+        filter((key) => Boolean(key))
+      )
+      .subscribe((id) =>
```

Comment:
You've introduced a memory leak.

Reply:
you probably better use the `key$` as an inner observable of the `treeAction$` so that you don't need to subscribe at all.